ARCHITECTURE.md

```diff
@@ -60,12 +52,43 @@ Rojo has many data structures and their purpose might not be immediately clear a
 
 To learn more, read about [`memofs` architecture](crates/memofs/ARCHITECTURE.md).
 
+### LiveServer
+
+LiveServer underlies the [`serve` command](#the-serve-command) and provides the web server which clients (such as the plugin) can use to interface with [`ServeSession`](#servesession).
+
+There's two parts to the API, the UI and the API clients use.
+
+The UI provides information about the current [`DOM`](#rojotree) has, including metadata. It also shows the project name, up-time, and version its Rojo is on.
+
+The API provides a simple JSON protocol to interact with and receive changes from the underlying [`ServeSession`](#servesession). Checkout the [`api.rs` file under the web module](src/web/api.rs) to learn more.
```

Comment:
This is fine for now, but we should eventually link this to the API documentation when it happens!

Reply:
Ah, I forgot that `doc.rust-lang.org` existed.

Reply:
Do you think we can use the [`web_api` crate doc page](https://docs.rs/rojo/latest/librojo/web_api/index.html) after brushing it up? If so, should we do it on this PR or a separate one?

Reply:
I think yes, though we should also try to have documentation for the web endpoints that aren't Rust-user focused. If someone from e.g. `roblox-ts` (this is an example, I don't know of anyone who does) wanted to use our endpoints, they shouldn't have to read through Rust docs to know what to expect and how to use them.